add more clarification regarding priorityClassName use with DaemonSet (#40851)
kubernetes/website: content/en/docs/concepts/workloads/controllers/daemonset.md

[priority](/docs/concepts/scheduling-eviction/pod-priority-preemption/#pod-priority)
of the new Pod.

> {{< note >}}
> If it's important that the DaemonSet pod run on each node, it's often desirable
> to set the `.spec.template.spec.priorityClassName` of the DaemonSet to a value
> like `system-cluster-critical` to ensure that this eviction occurs.
> {{< /note >}}

Reviewer: since PriorityClass isn't mentioned on this page before, how about adding a link for folks who are not familiar with PriorityClass
```suggestion
{{< note >}}
If it's important that the DaemonSet pod run on each node, it's often desirable
to set the `.spec.template.spec.priorityClassName` of the DaemonSet to an existing [PriorityClass](/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) 
like `system-cluster-critical` to ensure that this eviction occurs.
{{< /note >}}
```

Author: Done

> like `system-cluster-critical` to ensure that this eviction occurs.

Reviewer 2: Avoid recommending this PriorityClass. If we name it here, people will set that without thinking about what they are doing.
